src/VirtualShield.cpp

```diff
@@ -36,19 +36,25 @@
 #if defined(ARDUINO) && ARDUINO > 100
   // Define the serial port that is used to talk to the virtual shield.
   #define VIRTUAL_SERIAL_PORT0 Serial
+  
+  //Force a link to printf float functionality. There is a bug where some boards do not link this properly.
+  asm(".global _printf_float");
 
   // If it has dual serial ports (Leonardo), prefer the second one (pins 0, 1) for bluetooth.
-  #if defined(__AVR_ATmega32U4__)
+  #if defined(__AVR_ATmega32U4__) || defined(ARDUINO_SAMD_MKR1000) || defined(_VARIANT_ARDUINO_101_X_)
     #define VIRTUAL_SERIAL_PORT1 Serial1
     #define debugSerial
     #define debugSerialIn
+	#define PLATFORM_CAST(x) (const char*)(x)
```

Comment:
It is independent of the serial ports, but it is only applicable in the same cases as the serial port difference. I'll update the comment, but the #if condition is exactly the one we want